instrumentation/manticore/lib/opentelemetry/instrumentation/manticore/patches/response.rb

```diff
@@ -0,0 +1,104 @@
+  module Instrumentation
+    module Manticore
+      module Patches
+        # Module to prepend to RestClient::Request for instrumentation
+        module Response
+          # Hooks into the #.call function which is executed for any verbs like:
+          #   [:get, :post, :put, :options, :header]
+          # Executes for async and sync requests.
+          # It has direct to Manticore::Response methods and instance variables.
+          # Take great care of any unintentional mutations:
+          # [@request, @response, .on_complete] - are being invoked directly from Manticore::Response
+          def call
+            wrapped_request = OpenTelemetry::Instrumentation::Manticore::Util::WrappedRequest.new(@request)
+            attributes = request_attributes(wrapped_request).merge(OpenTelemetry::Common::HTTP::ClientContext.attributes)
+            span = tracer.start_span("HTTP #{wrapped_request.method}", attributes: attributes, kind: :client)
+            OpenTelemetry::Trace.with_span(span) do
+              OpenTelemetry.propagation.inject(wrapped_request)
+            end
+            on_complete do |response|
+              annotate_span_with_response!(span, response)
+            end
+            # .super() somehow becomes missing for async requests. Below works for sync and async requests.
+            self.method(:call).super_method.call
+          rescue ::Manticore::ManticoreException => e
+            OpenTelemetry.logger.debug("Errored during tracing: #{e}")
+            span.record_exception(e)
+          ensure
+            span.finish
+          end
+
+          private
+
+          # @param [OpenTelemetry::Trace::Tracer.start_span] span A running span that has not been completed
+          # @param [Manticore::Response] response A @response instance object after an outgoing call is attempted
+          def annotate_span_with_response!(span, response)
+            attr = response_attributes(response)
+            span.add_attributes(attr)
+            span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(response.code)
+          end
+
+          # @param [OpenTelemetry::Instrumentation::Manticore::Util::WrappedRequest] wrapped_request A WrappedRequest object that exposes some useful methods.
+          # @return [Hash] returns request attributes to be added to span
+          def request_attributes(wrapped_request)
+            uri = URI.parse(wrapped_request.uri)
+            attr = {
+              'library' => 'Manticore',
+              'http.method' => wrapped_request.method,
+              'http.scheme' => uri.scheme,
+              'http.target' => uri.path,
+              'http.url' => "#{uri.scheme}://#{uri.host}",
+              'net.peer.name' => uri.host,
+              'net.peer.port' => uri.port
+            }
+            return attr if Manticore::Instrumentation.instance.config["record_request_headers_list"].empty?
+            header_attr = header_attributes(wrapped_request.headers,
```

Comment:
small nite, The rack/request header list config option option i'd like to see follow the same config format as the `rack` name, just to keep folks mental model of config options consistent. https://github.com/open-telemetry/opentelemetry-ruby/blob/b7149fb72cad6c45956a6be80acfb2d062aab3a6/instrumentation/rack/lib/opentelemetry/instrumentation/rack/instrumentation.rb#L25-L26

Reply:
No problem